src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/RedditAnalyzer.scala

```diff
@@ -11,11 +8,12 @@ class RedditAnalyzer extends Analyzer[RedditObject] with Serializable
   with AnalysisDefaults.EnableAll[RedditObject] {
   override def toSchema(item: RedditObject, locationFetcher: LocationFetcher, imageAnalyzer: ImageAnalyzer): ExtendedDetails[RedditObject] = {
     ExtendedDetails(
-      id = randomUUID(),
-      createdAtEpoch = now.getEpochSecond,
+      id = item.data.id.getOrElse("").toString,
+      eventtime = item.data.created_utc.get.toLong,

```

Comment:
Yes, but extremely unlikely. I made most fields in `RedditObjectData` optional since different types of objects include different fields (and not using `Optional` was causing the JSON parser to throw). I have a feeling that other stream sources have the potential of exhibiting this problem during JSON parsing.

So, I would like to propose that we create a followup task in order to test by programmatically triggering these  `None`/`null` cases. This way we can ensure that bad/missing network data doesn't cause the entire pipleline to fail if such an exception occurs.